autoscale/src/test/java/com/sequenceiq/periscope/endpointtests/DistroXAutoScaleScalingActivityV1EndpointTest.java

```diff
@@ -0,0 +1,283 @@
+
+    @LocalServerPort
+    private int port;
+
+    @MockBean
+    private ResourceAuthorizationService resourceAuthorizationService;
+
+    @MockBean
+    private ClusterProxyConfigurationService clusterProxyConfigurationService;
+
+    @MockBean
+    private QuartzJobInitializer quartzJobInitializer;
+
+    @MockBean(name = "grpcUmsClient")
+    private GrpcUmsClient grpcUmsClient;
+
+    @Inject
+    private ClusterRepository clusterRepository;
+
+    @Inject
+    private ClusterPertainRepository clusterPertainRepsitory;
+
+    @MockBean
+    private ScalingActivityService scalingActivityService;
+
+    private DistroXAutoScaleScalingActivityV1Endpoint distroXAutoScaleScalingActivityV1Endpoint;
+
+    @Before
+    public void setup() {
+        distroXAutoScaleScalingActivityV1Endpoint = new AutoscaleUserCrnClientBuilder(String.format(SERVICE_ADDRESS, port))
+                .build().withCrn(TEST_USER_CRN).distroXAutoScaleScalingActivityV1Endpoint();
+
+        Cluster testCluster = new Cluster();
+        testCluster.setStackCrn(TEST_CLUSTER_CRN);
+        testCluster.setStackName(TEST_CLUSTER_NAME);
+        testCluster.setCloudPlatform("AWS");
+
+        ClusterPertain clusterPertain = new ClusterPertain();
+        clusterPertain.setTenant(TEST_ACCOUNT_ID);
+        clusterPertain.setWorkspaceId(TEST_WORKSPACE_ID);
+        clusterPertain.setUserId(TEST_USER_ID.toString());
+        clusterPertain.setUserCrn(TEST_USER_CRN);
+
+        UserManagementProto.User user = UserManagementProto.User.newBuilder()
+                .setCrn(TEST_USER_CRN).setEmail("dummyuser@cloudera.com").setUserId(TEST_USER_ID.toString()).build();
+        testCluster.setClusterPertain(
+                clusterPertainRepsitory.findFirstByUserCrn(clusterPertain.getUserCrn())
+                        .orElseGet(() -> clusterPertainRepsitory.save(clusterPertain)));
+        clusterRepository.save(testCluster);
+        when(grpcUmsClient.getUserDetails(anyString(), any())).thenReturn(user);
+        doNothing().when(resourceAuthorizationService).authorize(eq("crn:cdp:iam:us-west-1:accid:cluster:mockuser@cloudera.com"), any(), any());
+        when(clusterProxyConfigurationService.getClusterProxyUrl()).thenReturn(Optional.of("http://clusterproxy"));
+    }
+
+    @After
+    public void tearDown() {
+        clusterRepository.deleteAll();
+        clusterPertainRepsitory.deleteAll();
+    }
+
+    @Test
+    public void testGetAllScalingActivityUsingClusterName() {
+        Cluster testCluster = new Cluster();
+        testCluster.setStackCrn(TEST_CLUSTER_CRN);
+        testCluster.setStackName(TEST_CLUSTER_NAME);
+        testCluster.setCloudPlatform("AWS");
+
+        ScalingActivity scalingActivity = createScalingActivity(testCluster, ActivityStatus.METRICS_COLLECTION_SUCCESS, Instant.now().toEpochMilli());
+        doReturn(List.of(scalingActivity)).when(scalingActivityService).findAllForCluster(any(), any());
```

Comment:
I don't think stubs need to be added for full-fledged endpointTests. They should go ahead and persist data in the in-memory database and actually return the appropriate value automatically. Adding stubs to the class isn't required.

Reply:
Done